core/trace/src/main/scala/org/typelevel/otel4s/trace/SpanContext.scala

```diff
@@ -90,4 +90,20 @@ object SpanContext {
 
   def fromContext[F[_]](context: Vault): Option[SpanContext] =
     context.lookup(key)
+
+  final private[trace] val SampledMask = 1
+}
+
+private final case class SpanContextImpl(
+    traceId: ByteVector,
+    spanId: ByteVector,
+    flags: Byte,
+    isValid: Boolean,
+    isRemote: Boolean
+) extends SpanContext {
+
+  def samplingDecision: SamplingDecision =
+    SamplingDecision.fromBoolean(
+      (flags & SpanContext.SampledMask) == SpanContext.SampledMask
+    )
```

Comment:
@iRevive I'm a little confused if this is the right API to expose here.

I see `SamplingDecision` is described here, which is actually an enum with three states.
https://opentelemetry.io/docs/specs/otel/trace/sdk/#shouldsample

But this seems to be different from the `Sampled` flag, which is only a `Boolean` and is described here.
https://opentelemetry.io/docs/specs/otel/trace/sdk/#sampling

I see that the Java API exposes a `Boolean` on `SpanContext`, even though they also have a [`SamplingDecision` enum](https://github.com/open-telemetry/opentelemetry-java/blob/0a2dd9e569fae8d54014ac72faa4acde26762cd1/sdk/trace/src/main/java/io/opentelemetry/sdk/trace/samplers/SamplingDecision.java#L9).
https://github.com/typelevel/otel4s/blob/6bb372d639d04019bd6334d67c35599d2a6fce9f/java/trace/src/main/scala/org/typelevel/otel4s/java/trace/WrappedSpanContext.scala#L42-L43